projects/gateway2/proxy_syncer/proxy_syncer.go

```diff
@@ -371,6 +371,7 @@ func (s *ProxySyncer) Init(ctx context.Context, dbg *krt.DebugHandler) error {
 			Namespace: u.GetNamespace(),
 		}
 		glooUs.SetMetadata(&md)
+		glooUs.NamespacedStatuses = &u.Status
```

Comment:
Just so I understand, is this block of code intended to do something similar to https://github.com/solo-io/solo-kit/blob/main/pkg/api/v1/clients/kube/resource_client.go#L477 where we are converting the kubeUpstream?

Why do we just do this for Upstreams and not for all other gloo resources?

Reply:
yes exactly, it is a non-solo-kit bound way of doing the conversion from kube type to solo-kit type

>Why do we just do this for Upstreams and not for all other gloo resources?
We do it for the only other gloo type, AuthConfigs, below at https://github.com/solo-io/gloo/pull/10384/files#diff-c658efd653154e7eb520a5ea6c646cdd45af7ed0ac56224360affdf481a562c4R719

Other than that, RLCs are a skv2 type and don't have the same challenges